source/Server.Extensibility/Extensions/Infrastructure/Web/IExecuteWhenWebHostStarts.cs

```diff
@@ -0,0 +1,22 @@
+using System.Threading.Tasks;
+
+namespace Octopus.Server.Extensibility.Extensions.Infrastructure.Web
+{
+    public interface IExecuteWhenWebHostStarts
```

Comment:
Could we line up this interface with something like https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.ihostedservice?view=dotnet-plat-ext-3.1

So what I am saying is that we rename it to `IWebHostService` and only include 2 methods:
`Task Start()` and `Task Stop()`

I honestly don't think we need the **ing** and **ed**.
In the implementation (which as far as I know is the only one) [we don't use `OnWebHostStarting`](https://github.com/OctopusDeploy/DirectoryServicesAuthenticationProvider/commit/d852aa2a026bf6b9818359100bc907f2eaa3dd05#diff-114e500eacbbdf6e0bd3edfa1eaf6c4dR36). 

I vote to keep it simple and line up with asp.net core already existing interfaces so when we actually go full asp.net core, the migration is simpler.